runtime/lua/vim/lsp.lua

```diff
@@ -2386,7 +2386,7 @@ function lsp.formatexpr(opts)
       }
       local response =
         client.request_sync('textDocument/rangeFormatting', params, timeout_ms, bufnr)
-      if response.result then
+      if response and response.result then
```

Comment:
this looks like a bug fix, not a lint change

Reply:
warnning about `need check nil` here just work around.